src/blocks/sound/pulseaudio.rs

```diff
@@ -182,78 +198,106 @@ impl Connection {
 impl Client {
     fn new() -> Result<Client> {
         let (send_req, recv_req) = unbounded();
+        let (wake_rx, wake_tx) = nix::unistd::pipe().unwrap();
+
+        extern "C" fn wake_cb(
```

Comment:
Could you add some comments from here through line 210? I found it pretty difficult to appreciate what's happening.

Reply:
Yeah, sure! I will probably refactor this code into something like `struct MainloopWaker`.